src/overlays/actors/ovl_En_Insect/z_en_insect.c

```diff
@@ -255,37 +275,36 @@ void func_80A7C3F4(EnInsect* this, GlobalContext* globalCtx) {
 
     SkelAnime_Update(&this->skelAnime);
     this->actor.shape.rot.y = this->actor.world.rot.y;
-    if (this->unk_31A <= 0) {
-        func_80A7C598(this);
+    if (this->actionTimer <= 0) {
+        EnInsect_SetupCrawl(this);
     }
 
-    if (((this->unk_314 & 4) && this->unk_31C <= 0) ||
-        ((sp2E == 2 || sp2E == 3) && (this->unk_314 & 1) && (this->actor.bgCheckFlags & BGCHECKFLAG_GROUND) &&
-         D_80A7DEB8 >= 4)) {
-        func_80A7CBC8(this);
-    } else if ((this->unk_314 & 1) && (this->actor.bgCheckFlags & BGCHECKFLAG_WATER_TOUCH)) {
-        func_80A7CE60(this);
+    if (((this->insectFlags & INSECT_FLAG_TEMP) && this->lifeTimer <= 0) ||
+        (IS_DROPPED(type) && (this->insectFlags & INSECT_FLAG_TEMP_AND_ALIVE) && (this->actor.bgCheckFlags & BGCHECKFLAG_WATER_TOUCH) && sDroppedCount >= 4)) {
```

Comment:
merge issue here, I think that is the only issue
```suggestion
        (IS_DROPPED(type) && (this->insectFlags & INSECT_FLAG_TEMP_AND_ALIVE) && (this->actor.bgCheckFlags & BGCHECKFLAG_GROUND) && sDroppedCount >= 4)) {
```